synapse/handlers/room.py

```diff
@@ -871,7 +881,9 @@ async def _send_events_for_new_room(
             The stream_id of the last event persisted.
         """
 
-        creator_id = creator.user.to_string()
+        if not creator:
+            creator = requester.user
```

Comment:
this looks redundant, since we're also doing it in `create_room`.  I think we should make `creator` mandatory for `_send_events_for_new_room`.